src/storage-service.ts

```diff
@@ -13,16 +13,20 @@ export class MockStorageService implements StorageService {
   readonly proofSetId: ProofSetId
   readonly storageProvider: StorageProvider
   private readonly _storedData: Map<string, Uint8Array> = new Map()
+  private readonly _signerAddress: string
+  private readonly _withCDN: boolean
 
-  constructor (proofSetId: ProofSetId, storageProvider: StorageProvider) {
+  constructor (proofSetId: ProofSetId, storageProvider: StorageProvider, signerAddress: string, withCDN: boolean) {
```

Comment:
Boolean function arguments are difficult to understand when reading the code.

```js
new MockStorageService(proofSetId, storageProvider, address, true)
//                 It's not clear what does "true" mean here? ^^^
```

I propose using an "options" object.

```suggestion
  constructor (proofSetId: ProofSetId, storageProvider: StorageProvider, signerAddress: string, options: { withCDN?: boolean } = {}) {
```

Reply:
https://github.com/FilOzone/synapse-sdk/blob/0a42daac1205d13ca22a86f7e5d1a9fba1618f2e/src/synapse.ts#L137-L142

Reply:
This is a private constructor so I like being explicit here and let the factory method do the validating. With enough arguments this is going to get unwieldily, but it's still private so that's not a user concern.

But I'm easy on this tbh, I like the pattern of validating flexible args in a factory and be explicit in a constructor, and the factory only exists at this point to deal with async awkwardness. As long as properties get set and we have some consistency.